### ISSUE TITLE
jupyter lab --watch is broken in v2.1.3

### ISSUE THREAD

**Issue by ianhi** [jupyterlab/jupyterlab#8469]
## Description
I was trying to create a new conda environment for working on an extension and running `jupyter lab --watch` resulted in this error:
```
Error: No version of chokidar is available. Tried chokidar@2 and chokidar@3.
You could try to manually install any chokidar version.
chokidar@3: Error: Cannot find module 'chokidar'
Require stack:
```

<details><summary>Full terminal output</summary>
<pre>
ian@pop-os:~$ conda create -n jlab-testv3 --override-channels --strict-channel-priority -c conda-forge -c anaconda jupyterlab=2.1.3 nodejs
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/ian/anaconda3/envs/jlab-testv3

  added / updated specs:
    - jupyterlab=2.1.3
    - nodejs

The following NEW packages will be INSTALLED:

  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge
  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-0_gnu
  attrs              conda-forge/noarch::attrs-19.3.0-py_0
  backcall           conda-forge/noarch::backcall-0.1.0-py_0
  binutils_impl_lin~ conda-forge/linux-64::binutils_impl_linux-64-2.34-h53a641e_2
  bleach             conda-forge/noarch::bleach-3.1.5-pyh9f0ad1d_0
  brotlipy           conda-forge/linux-64::brotlipy-0.7.0-py38h1e0a361_1000
  ca-certificates    conda-forge/linux-64::ca-certificates-2020.4.5.1-hecc5488_0
  certifi            conda-forge/linux-64::certifi-2020.4.5.1-py38h32f6830_0
  cffi               conda-forge/linux-64::cffi-1.14.0-py38hd463f26_0
  chardet            conda-forge/linux-64::chardet-3.0.4-py38h32f6830_1006
  cryptography       conda-forge/linux-64::cryptography-2.9.2-py38h766eaa4_0
  decorator          conda-forge/noarch::decorator-4.4.2-py_0
  defusedxml         conda-forge/noarch::defusedxml-0.6.0-py_0
  entrypoints        conda-forge/linux-64::entrypoints-0.3-py38h32f6830_1001
  gcc_impl_linux-64  conda-forge/linux-64::gcc_impl_linux-64-7.5.0-hd420e75_6
  gfortran_impl_lin~ conda-forge/linux-64::gfortran_impl_linux-64-7.5.0-hdf63c60_6
  gxx_impl_linux-64  conda-forge/linux-64::gxx_impl_linux-64-7.5.0-hdf63c60_6
  icu                conda-forge/linux-64::icu-67.1-he1b5a44_0
  idna               conda-forge/noarch::idna-2.9-py_1
  importlib-metadata conda-forge/linux-64::importlib-metadata-1.6.0-py38h32f6830_0
  importlib_metadata conda-forge/noarch::importlib_metadata-1.6.0-0
  ipykernel          conda-forge/linux-64::ipykernel-5.3.0-py38h23f93f0_0
  ipython            conda-forge/linux-64::ipython-7.14.0-py38h32f6830_0
  ipython_genutils   conda-forge/noarch::ipython_genutils-0.2.0-py_1
  jedi               conda-forge/linux-64::jedi-0.17.0-py38h32f6830_0
  jinja2             conda-forge/noarch::jinja2-2.11.2-pyh9f0ad1d_0
  json5              conda-forge/noarch::json5-0.9.0-py_0
  jsonschema         conda-forge/linux-64::jsonschema-3.2.0-py38h32f6830_1
  jupyter_client     conda-forge/noarch::jupyter_client-6.1.3-py_0
  jupyter_core       conda-forge/linux-64::jupyter_core-4.6.3-py38h32f6830_1
  jupyterlab         conda-forge/noarch::jupyterlab-2.1.3-py_0
  jupyterlab_server  conda-forge/noarch::jupyterlab_server-1.1.5-py_0
  ld_impl_linux-64   conda-forge/linux-64::ld_impl_linux-64-2.34-h53a641e_2
  libffi             conda-forge/linux-64::libffi-3.2.1-he1b5a44_1007
  libgcc-ng          conda-forge/linux-64::libgcc-ng-9.2.0-h24d8f2e_2
  libgfortran-ng     conda-forge/linux-64::libgfortran-ng-7.5.0-hdf63c60_6
  libgomp            conda-forge/linux-64::libgomp-9.2.0-h24d8f2e_2
  libsodium          conda-forge/linux-64::libsodium-1.0.17-h516909a_0
  libstdcxx-ng       conda-forge/linux-64::libstdcxx-ng-9.2.0-hdf63c60_2
  libuv              conda-forge/linux-64::libuv-1.34.0-h516909a_0
  markupsafe         conda-forge/linux-64::markupsafe-1.1.1-py38h1e0a361_1
  mistune            conda-forge/linux-64::mistune-0.8.4-py38h1e0a361_1001
  nbconvert          conda-forge/linux-64::nbconvert-5.6.1-py38h32f6830_1
  nbformat           conda-forge/noarch::nbformat-5.0.6-py_0
  ncurses            conda-forge/linux-64::ncurses-6.1-hf484d3e_1002
  nodejs             conda-forge/linux-64::nodejs-14.3.0-h568c755_0
  notebook           conda-forge/linux-64::notebook-6.0.3-py38h32f6830_0
  openssl            conda-forge/linux-64::openssl-1.1.1g-h516909a_0
  packaging          conda-forge/noarch::packaging-20.4-pyh9f0ad1d_0
  pandoc             conda-forge/linux-64::pandoc-2.9.2.1-0
  pandocfilters      conda-forge/noarch::pandocfilters-1.4.2-py_1
  parso              conda-forge/noarch::parso-0.7.0-pyh9f0ad1d_0
  pexpect            conda-forge/linux-64::pexpect-4.8.0-py38h32f6830_1
  pickleshare        conda-forge/linux-64::pickleshare-0.7.5-py38h32f6830_1001
  pip                conda-forge/noarch::pip-20.1.1-pyh9f0ad1d_0
  prometheus_client  conda-forge/noarch::prometheus_client-0.7.1-py_0
  prompt-toolkit     conda-forge/noarch::prompt-toolkit-3.0.5-py_0
  ptyprocess         conda-forge/noarch::ptyprocess-0.6.0-py_1001
  pycparser          conda-forge/noarch::pycparser-2.20-py_0
  pygments           conda-forge/noarch::pygments-2.6.1-py_0
  pyopenssl          conda-forge/noarch::pyopenssl-19.1.0-py_1
  pyparsing          conda-forge/noarch::pyparsing-2.4.7-pyh9f0ad1d_0
  pyrsistent         conda-forge/linux-64::pyrsistent-0.16.0-py38h1e0a361_0
  pysocks            conda-forge/linux-64::pysocks-1.7.1-py38h32f6830_1
  python             conda-forge/linux-64::python-3.8.2-he5300dc_7_cpython
  python-dateutil    conda-forge/noarch::python-dateutil-2.8.1-py_0
  python_abi         conda-forge/linux-64::python_abi-3.8-1_cp38
  pyzmq              conda-forge/linux-64::pyzmq-19.0.1-py38ha71036d_0
  readline           conda-forge/linux-64::readline-8.0-hf8c457e_0
  requests           conda-forge/noarch::requests-2.23.0-pyh8c360ce_2
  send2trash         conda-forge/noarch::send2trash-1.5.0-py_0
  setuptools         conda-forge/linux-64::setuptools-46.4.0-py38h32f6830_0
  six                conda-forge/noarch::six-1.15.0-pyh9f0ad1d_0
  sqlite             conda-forge/linux-64::sqlite-3.30.1-hcee41ef_0
  sysroot_linux-64   conda-forge/noarch::sysroot_linux-64-2.12-h3a0023d_3
  terminado          conda-forge/linux-64::terminado-0.8.3-py38h32f6830_1
  testpath           conda-forge/noarch::testpath-0.4.4-py_0
  tk                 conda-forge/linux-64::tk-8.6.10-hed695b0_0
  tornado            conda-forge/linux-64::tornado-6.0.4-py38h1e0a361_1
  traitlets          conda-forge/linux-64::traitlets-4.3.3-py38h32f6830_1
  urllib3            conda-forge/noarch::urllib3-1.25.9-py_0
  wcwidth            conda-forge/noarch::wcwidth-0.1.9-pyh9f0ad1d_0
  webencodings       conda-forge/noarch::webencodings-0.5.1-py_1
  wheel              conda-forge/noarch::wheel-0.34.2-py_1
  xz                 conda-forge/linux-64::xz-5.2.5-h516909a_0
  zeromq             conda-forge/linux-64::zeromq-4.3.2-he1b5a44_2
  zipp               conda-forge/noarch::zipp-3.1.0-py_0
  zlib               conda-forge/linux-64::zlib-1.2.11-h516909a_1006

Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate jlab-testv3
#
# To deactivate an active environment, use
#
#     $ conda deactivate

ian@pop-os:~$ conda activate jlab-testv3
(jlab-testv3) ian@pop-os:~$ jupyter lab --watch
[I 16:32:45.206 LabApp] The port 8888 is already in use, trying another port.
[I 16:32:45.489 LabApp] JupyterLab extension loaded from /home/ian/anaconda3/envs/jlab-testv3/lib/python3.8/site-packages/jupyterlab
[I 16:32:45.489 LabApp] JupyterLab application directory is /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab
[I 16:32:45.489 LabApp] Starting JupyterLab watch mode...
[I 16:32:52.028 LabApp] > node /home/ian/anaconda3/envs/jlab-testv3/lib/python3.8/site-packages/jupyterlab/staging/yarn.js run watch
yarn run v1.21.1
$ webpack --watch

webpack is watching the files…

/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/watchpack/lib/chokidar.js:17
throw new Error(
^

Error: No version of chokidar is available. Tried chokidar@2 and chokidar@3.
You could try to manually install any chokidar version.
chokidar@3: Error: Cannot find module 'chokidar'
Require stack:
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/watchpack/lib/chokidar.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/watchpack/lib/DirectoryWatcher.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/watchpack/lib/watcherManager.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/watchpack/lib/watchpack.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack/lib/node/NodeWatchFileSystem.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack/lib/node/NodeEnvironmentPlugin.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack/lib/webpack.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack-cli/bin/utils/validate-options.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack-cli/bin/utils/convert-argv.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack-cli/bin/cli.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack/bin/webpack.js
chokidar@2: Error: Cannot find module 'watchpack-chokidar2'
Require stack:
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/watchpack/lib/chokidar.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/watchpack/lib/DirectoryWatcher.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/watchpack/lib/watcherManager.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/watchpack/lib/watchpack.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack/lib/node/NodeWatchFileSystem.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack/lib/node/NodeEnvironmentPlugin.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack/lib/webpack.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack-cli/bin/utils/validate-options.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack-cli/bin/utils/convert-argv.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack-cli/bin/cli.js
- /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack/bin/webpack.js

    at Object.<anonymous> (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/watchpack/lib/chokidar.js:17:7)
    at Module._compile (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1220:10)
    at Module.load (internal/modules/cjs/loader.js:1049:32)
    at Function.Module._load (internal/modules/cjs/loader.js:937:14)
    at Module.require (internal/modules/cjs/loader.js:1089:19)
    at require (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/v8-compile-cache/v8-compile-cache.js:161:20)
    at Object.<anonymous> (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/watchpack/lib/DirectoryWatcher.js:9:16)
    at Module._compile (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1220:10)
    at Module.load (internal/modules/cjs/loader.js:1049:32)
    at Function.Module._load (internal/modules/cjs/loader.js:937:14)
    at Module.require (internal/modules/cjs/loader.js:1089:19)
    at require (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/v8-compile-cache/v8-compile-cache.js:161:20)
    at WatcherManager.getDirectoryWatcher (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/watchpack/lib/watcherManager.js:14:25)
    at WatcherManager.watchFile (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/watchpack/lib/watcherManager.js:28:14)
    at Watchpack.<anonymous> (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/watchpack/lib/watchpack.js:38:49)
    at Array.map (<anonymous>)
    at Watchpack.watch (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/watchpack/lib/watchpack.js:37:28)
    at NodeWatchFileSystem.watch (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack/lib/node/NodeWatchFileSystem.js:70:16)
    at Watching.watch (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack/lib/Watching.js:114:48)
    at /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack/lib/Watching.js:101:10
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at AsyncSeriesHook.lazyCompileHook (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/tapable/lib/Hook.js:154:20)
    at Watching._done (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack/lib/Watching.js:98:28)
    at /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack/lib/Watching.js:73:19
    at Compiler.emitRecords (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack/lib/Compiler.js:499:39)
    at /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack/lib/Watching.js:54:20
    at /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack/lib/Compiler.js:485:14
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at AsyncSeriesHook.lazyCompileHook (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/tapable/lib/Hook.js:154:20)
    at /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack/lib/Compiler.js:482:27
    at /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/neo-async/async.js:2818:7
    at done (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/neo-async/async.js:3522:9)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at /home/ian/anaconda3/envs/jlab-testv3/share/jupyter/lab/staging/node_modules/webpack/lib/Compiler.js:464:33
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
</pre>
</details>

This error happens for v2.1.3 but not for v2.1.2

## Reproduce
```bash
conda create -n jlab-testv3 --override-channels --strict-channel-priority -c conda-forge -c anaconda jupyterlab=2.1.3 nodejs
conda activate jlab-testv3
jupyter lab --watch
```
- Operating System and version: PopOs 20.04
- Browser and version: N/A
- JupyterLab version: 2.1.3

## Expected behavior

This is fixable with `npm install chokidar -g` but ideally the previous behavior where I don't have to install chokidar manually would be restored

**Comment by lykqnbz** [comment]
Thanks a million

**Comment by blink1073** [comment]
@ianhi, if you make `chokidar` a `devDependency` in [`dev_mode`](https://github.com/jupyterlab/jupyterlab/blob/b62d4fd463c6d00451df6327203413cf1b520a8f/dev_mode/package.json#L61), I can backport and make a 2.1.x release.

**Comment by ianhi** [comment]
Does the `dev_mode` folder apply when not using the `--dev-mode` flag? Also, I noticed that chokidar doesn't seem to have been included as a dev dependency in the past, see: https://github.com/jupyterlab/jupyterlab/blob/2.0.x/dev_mode/package.json

**Comment by blink1073** [comment]
Dev_mode files are copied to the static files on release.  My guess is that chokidar was picked up as a dependency of a dependency before by happenstance.

**Comment by jtpio** [comment]
Just faced the same issue on a third-party extension, and opened https://github.com/jupyterlab/jupyterlab/pull/8481

**Comment by ianhi** [comment]
Thanks! With that merged I'll close this

**Comment by ianhi** [comment]
Oops, I suppose I should wait to close until successfully backported - reopening

**Comment by ianhi** [comment]
Working in the now published 2.1.4 https://github.com/jupyterlab/jupyterlab/commit/3ce3b9f90bf461e3ec15787ae97b0dbdfca31144

Thanks!
